aadhaar_demo_auth/data_store.py

```diff
@@ -37,17 +38,19 @@ def ensure_dir(self):
 
     def store_raw_request(self, root_node):
         self.ensure_dir()
-        with open(self.raw_request_file(), 'w+') as f:
-            f.write(etree.tostring(root_node, encoding='UTF-8',
-                                   xml_declaration=True, pretty_print=True))
+        logging.basicConfig(filename=self.raw_request_file(), level=logging.DEBUG)
+        logging.info(etree.tostring(
+            root_node, encoding='UTF-8',
+            xml_declaration=True, pretty_print=True))
 
     def store_signed_request(self, root_node):
         self.ensure_dir()
-        with open(self.signed_request_file(), 'w+') as f:
-            f.write(etree.tostring(root_node, encoding='UTF-8',
-                                   xml_declaration=True, pretty_print=True))
+        logging.basicConfig(filename=self.signed_request_file(), level=logging.DEBUG)
+        logging.info(etree.tostring(
+            root_node, encoding='UTF-8',
+            xml_declaration=True, pretty_print=True))
 
     def store_response(self, xml_content):
         self.ensure_dir()
-        with open(self.signed_request_file(), 'w+') as f:
-            f.write(xml_content)
+        logging.basicConfig(filename=self.response_file(), level=logging.DEBUG)
+        logging.info(xml_content)
```

Comment:
Whenever you log a response or request or whatever, this makes sure the directory exists, reconfigures the logger to use the appropriate log file, then logs the message.  The first two parts of that should happen just once as part of a logger setup.

I'm not very familiar with the python logging module, but I believe it does handle all this sort of thing already.  I think you could swap out this whole class with standard library loggers (except for maybe the `etree` stuff).  What that would look like is a localsettings file or something where you specify the log directory, then something like:
```python
signed_request_logger = logging.getLogger('signed_request')
```
And then I think you add a handler to it or something where you pass the filename to log to.  Not sure exactly.